AK/BigInts.h

```diff
@@ -0,0 +1,487 @@
+    {
+        return !m_high && m_low < other;
+    }
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr bool operator>=(const T& other) const
+    {
+        return *this == other || *this > other;
+    }
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr bool operator<=(const T& other) const
+    {
+        return *this == other || *this < other;
+    }
+
+    constexpr bool operator==(const R& other) const
+    {
+        return m_low == other.low() && m_high == other.high();
+    }
+    constexpr bool operator!=(const R& other) const
+    {
+        return m_low != other.low() || m_high != other.high();
+    }
+    constexpr bool operator>(const R& other) const
+    {
+        return m_high > other.high()
+            || (m_high == other.high() && m_low > other.low());
+    }
+    constexpr bool operator<(const R& other) const
+    {
+        return m_high < other.high()
+            || (m_high == other.high() && m_low < other.low());
+    }
+    constexpr bool operator>=(const R& other) const
+    {
+        return *this == other || *this > other;
+    }
+    constexpr bool operator<=(const R& other) const
+    {
+        return *this == other || *this < other;
+    }
+
+    // Bitwise operations
+    constexpr R&& operator~()
+    {
+        return { ~m_low, ~m_high };
+    }
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr U&& operator&(const U& other) const
+    {
+        return static_cast<U>(m_low) & other;
+    }
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr R&& operator|(const U& other) const
+    {
+        return { m_low | other, m_high };
+    }
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr R&& operator^(const U& other) const
+    {
+        return { m_low ^ other, m_high };
+    }
+    template<Unsigned U>
+    constexpr R&& operator<<(const U& other) const
+    {
+        if (other > sizeof(this))
+            return 0;
+
+        u64 overflow = m_low >> (sizeof(T) - other);
+        return { m_low << other, (m_high << other) | overflow };
+    }
+    template<Unsigned U>
+    constexpr R&& operator>>(const U& other) const
+    {
+        if (other > sizeof(this))
+            return 0;
+
+        T underflow = (m_high << (sizeof(T) - other)) >> other;
+        return { (m_low >> other) | (underflow << (sizeof(T) - other)), m_high >> other };
+    }
+
+    constexpr R&& operator&(const R& other) const
+    {
+        return { m_low & other.low(), m_high & other.high() };
+    }
+    constexpr R&& operator|(const R& other) const
+    {
+        return { m_low | other.low(), m_high | other.high() };
+    }
+    constexpr R&& operator^(const R& other) const
+    {
+        return { m_low ^ other.low(), m_high ^ other.high() };
+    }
+
+    // Bitwise assignment
+    template<Unsigned U, EnableIf<sizeof(U) <= sizeof(T)>::Type>
+    constexpr R& operator&=(const U& other)
```

Comment:
I mentioned this before, but it'd be really nice if we could move away from SFINAE where possible, due to having less ugly compile errors, if nothing else

```suggestion
    constexpr R& operator&=(const U& other) requires(sizeof(U) <= sizeof(T))
```

Reply:
yeah, I was to stupid to use requires and at that point EnableIf's were easier, I will probably rework all these EnableIfs into requirements